### PR TITLE
Block TLS packet rather than SYN packet

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/1202279501986195/1203277901387351/f

### Description
We would previously use the SYN packet as a place to decide whether we want to block a request or not.
That had some nuances as described in [this task](https://app.asana.com/0/1202279501986195/1203277901387351/f)

This makes some changes so that the place to decide whether to block a request or not is the TLS handshake instead.

### Steps to test this PR
Tests from https://github.com/duckduckgo/Android/pull/2482

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5** <a href="https://app.graphite.dev/github/pr/duckduckgo/netguard/5" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/netguard/5?utm_source=stack-comment).

File: src/tls.c [hunk 1]
```diff
@@ -41,12 +42,12 @@ void tls_sni_inspection(
...
     if (strlen(sn) == 0) {
         log_print(PLATFORM_LOG_PRIORITY_INFO, "TLS server name not found");
-        return;
+        return 0;
     }
 
     log_print(PLATFORM_LOG_PRIORITY_INFO, "TLS server %s (%s) found", sn, dest);
 
-    sni_resolved(args, sn, dest);
+    return is_domain_blocked(args, sn, uid);
```
Comment (nshuba) [inline src/tls.c:50]: This is just a nit, but we're mixing `jboolean` and `int` here. Would be good to clean up at some point (does not have to be now).

**Review by nshuba — APPROVED** [review]
LGTM.

File: src/tls.c (hunk 1 above)
Comment (aitorvs) [reply src/tls.c:50]: Yeah, I am planning a general clean up plus testing...sigh!

**Pull request merged**